src/Umbraco.Web/ContentApps/ContentAppDefinitionCollectionBuilder.cs

```diff
@@ -25,14 +27,14 @@ public override ContentAppDefinitionCollection CreateCollection()
             return new ContentAppDefinitionCollection(CreateItems(), logger);
         }
 
-        protected override IEnumerable<IContentAppDefinition> CreateItems(params object[] args)
+        protected override IEnumerable<IContentAppFactory> CreateItems(params object[] args)
         {
             // get the manifest parser just-in-time - injecting it in the ctor would mean that
             // simply getting the builder in order to configure the collection, would require
             // its dependencies too, and that can create cycles or other oddities
             var manifestParser = Container.GetInstance<ManifestParser>();
 
```

Comment:
It's here the plan is to inject the `IContentTypeService` and pass into the `ManifestContentAppFactory` to handle the "magic"